src/uwtools/tests/test_schemas.py

```diff
@@ -297,6 +297,29 @@ def ww3_prop():
     return partial(schema_validator, "ww3", "properties", "ww3", "properties")
 
 
+# Helpers
+
+
+def non_empty_dict(errors: list[str]) -> bool:
+    for msg in [
+        "{} does not have enough properties",  # jsonschema [4.18,4.20.*]
+        "{} should be non-empty",  # jsonschema [4.21.*,?]
+    ]:
+        if msg in errors:
+            return True
+    return False
+
+
+def non_empty_list(errors: list[str]) -> bool:
+    for msg in [
+        "[] is too short",  # jsonschema [4.18.*,4.20.*]
+        "[] should be non-empty",  # jsonschema [4.21.*,?]
+    ]:
+        if msg in errors:
+            return True
+    return False
```

Comment:
If I'd found a way to detect the version of `jsonschema` currently installed, I would have made the expected error message conditional on that. But I couldn't find a straightforward way to detect the version at run time.